File: MATLAB/NewtonRaphsonModMet.m
% NewtonRaphson con una modificación
function [mNM]=NewtonRaphsonModMet(x0,f,tol,iterMax)
  % Declaracion de variables
  iter = 1:iterMax;                           % Definir iter como un vector
  df = diff(f);                               % Obtener funcion derivada de f
  err = tol + 1e0;                            % Error inicia sobre tol
  U = f/df;
  dU = diff(U);
  % Ciclo de iteraciones
  for i = iter
    if (subs(dU,x0) == 0)                     % Verificar si hay divergencia
      disp("Diverge");
      break
    elseif (vpa(err) <= vpa(tol))             % Verificar si se llego a tol
      disp("Maxima tolerancia alcanzada");
      break      
    else                                      % Proceder a iteracion
      xi = vpa(x0 - subs(U,x0)/subs(dU,x0));  % Calcular nueva x
      err = abs(x0 - xi);                     % Obtener error absoluto
      x0 = xi;                                % Redefinir x0 como xi
      mNM(i,1) = fix(vpa(i));
      mNM(i,2) = xi;                          % Almacenar en mNR
      mNM(i,3) = err;
    end
  end
end

File: Python/NewtonRaphsonModMet.py
import sympy as sym

def newton_raphson_mod_met(x0,f,tol,iter_max):
    # Variable declaration
    m_nr = []
    x = sym.Symbol('x')
    iter = range(iter_max)
    df = sym.diff(f)
    err = tol + 1
    u = f/df
    du = sym.diff(u)
    # Iteration cycle
    for i in iter:
        if du.subs(x,x0) == 0:
            print("Detected divergence")
            break
        elif err <= tol:
            print("Tolerance limit reached")
            break
        else:
            xi = x0 - u.subs(x,x0)/du.subs(x,x0)
            err = abs(x0 - xi)
            x0 = xi
            m_nr.append([i,float(xi),float(err)])
    return(m_nr)
